### PR TITLE
Automatically register SpecFlow bindings

### LINKED ISSUE #7: Automatically register SpecFlow bindings
Currently the developer has to write a little bit of setup code to setup the DI for SpecFlow (as described in the [README.md](/solidtoken/SpecFlow.DependencyInjection/blob/master/README.md)). This setup code needs to register all the SpecFlow ``` [Binding]```s otherwise the SpecFlow tests will fail with a obscure error message.

As a developer, I want to automatically register all ```[Binding]```s in the same Assembly where the ```[ScenarioDependencies]``` is defined.

- [x] Add a ```AutoRegisterBindings``` (default True) to ```[ScenarioDependencies```
- [x] Automatically register the appropriate ```[Binding]```s

### LINKED ISSUE #24: System.NullReferenceException inside TechTalk.SpecFlow
Hi,

I added this to my specflow project:

```xml
    <PackageReference Include="SpecFlow" Version="3.1.80" />
    <PackageReference Include="SpecFlow.MsTest" Version="3.1.80" />
    <PackageReference Include="SpecFlow.Tools.MsBuild.Generation" Version="3.1.80" />
    <PackageReference Include="SolidToken.SpecFlow.DependencyInjection" Version="0.3.2" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.4.0" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.0.0" />
    <PackageReference Include="MSTest.TestFramework" Version="2.0.0" />
```

and 

```c#
	public class SpecflowHooks
	{
		
		[ScenarioDependencies]
		public static IServiceCollection CreateServices()
		{
			var services = new ServiceCollection();
			return services;
		}
	}
```

And now I get for all tests:
```
System.NullReferenceException: Object reference not set to an instance of an object.
    at lambda_method(Closure , IContextManager , String )
   at TechTalk.SpecFlow.Bindings.BindingInvoker.InvokeBinding(IBinding binding, IContextManager contextManager, Object[] arguments, ITestTracer testTracer, TimeSpan& duration) in D:\a\1\s\TechTalk.SpecFlow\Bindings\BindingInvoker.cs:line 69
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStepMatch(BindingMatch match, Object[] arguments) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 501
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStep(IContextManager contextManager, StepInstance stepInstance) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 422
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.Step(StepDefinitionKeyword stepDefinitionKeyword, String keyword, String text, String multilineTextArg, Table tableArg) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 581
   at TechTalk.SpecFlow.TestRunner.Given(String text, String multilineTextArg, Table tableArg, String keyword) in D:\a\1\s\TechTalk.SpecFlow\TestRunner.cs:line 80
   at 
```

any idea?

Update: same issue with 0.3.1 and 0.2.4. What am I missing?

### PR DESCRIPTION
**Pull request by 304NotModified**
Fixes #24, fixes #7

### CODE REVIEW

**Comment by 304NotModified** [comment]
Please squash when accepted :)

**Comment by mbhoek** [comment]
You move fast! 😄 Let me check this tonight, I think there was a reason for using the `return () =>` and `Func<IServiceCollection>` constructs having to do with timing so I just want to make sure.

Thanks for your contribution!

**Comment by 304NotModified** [comment]
it's still doing `return () => ` ;)

But please do check :)

Thanks for the great library!

**Comment by 304NotModified** [comment]
@mbhoek I've updated the PR so it also resolves #7 :)

**Review by mbhoek — COMMENTED** [review]
Looks ok, but I need to check what happens if users already added the piece of initialisation code (and this change would add those bindings again). This comment is a reminder to @mbhoek to check it.

EDIT: and ofcourse simultaneously @304NotModified updated the PR, so I need to recheck 😃

**Comment by 304NotModified** [comment]
luckily I didn't do a force push, so you could check the latest commit :)

**Comment by 304NotModified** [comment]
> Looks ok, but I need to check what happens if users already added the piece of initialisation code

No crash at my side, but not sure if there goes some else wrong

**Comment by 304NotModified** [comment]
To be clear, if you leave this:

```c#
            foreach (var type in typeof(TestDependencies).Assembly.GetTypes().Where(t => Attribute.IsDefined(t, typeof(BindingAttribute))))	
            {	
                services.AddSingleton(type);	
            }
```

Then still all tests are green.

File: SpecFlow.DependencyInjection/ScenarioDependenciesAttribute.cs [hunk 1]
```diff
@@ -5,5 +5,9 @@ namespace SolidToken.SpecFlow.DependencyInjection
     [AttributeUsage(AttributeTargets.Method)]
     public class ScenarioDependenciesAttribute : Attribute
     {
+        /// <summary>
+        /// Automatically register all SpecFlow bindings
```
Comment (mbhoek) [inline SpecFlow.DependencyInjection/ScenarioDependenciesAttribute.cs:9]: Prefer ending all `<summary>` comments with .

**Review by mbhoek — APPROVED** [review]
👍

**Pull request merged**